ts/main.ts

```diff
@@ -36,24 +38,53 @@ function previewPhoto(event: Event): void {
   $imgPreview.setAttribute('src', eventTarget.value);
 }
 
+function previewPhotoByString(event: string): void {
+  if (!$imgPreview) throw new Error('$imgPreview not found!');
+  $imgPreview.setAttribute('src', event);
+}
+
 if (!$form) throw new Error('$form not found!');
 $form.addEventListener('submit', submitHandler);
 
 function submitHandler(event: Event): void {
   event.preventDefault();
   if (!$form) throw new Error('$form not found!');
   const elements = $form.elements as JournalEntryPieces;
-  const newEntry: Entry = {
-    title: elements.title.value,
-    imgUrl: elements.photo.value,
-    notes: elements.notes.value,
-    entryId: data.nextEntryId,
-  };
-  data.entries.unshift(newEntry);
-  data.nextEntryId++;
-  const $liToAppend = renderEntry(data.entries[0]);
-  if (!$ulForEntries) throw new Error('$ulForEntries not found!');
-  $ulForEntries.prepend($liToAppend);
+  let newEntry: Entry;
+  if (data.editing === null) {
+    newEntry = {
+      title: elements.title.value,
+      imgUrl: elements.photo.value,
+      notes: elements.notes.value,
+      entryId: data.nextEntryId,
+    };
+    data.entries.unshift(newEntry);
+    data.nextEntryId++;
+    const $liToAppend = renderEntry(newEntry);
+    if (!$ulForEntries) throw new Error('$ulForEntries not found!');
+    $ulForEntries.prepend($liToAppend);
+  } else {
+    newEntry = {
+      title: elements.title.value,
+      imgUrl: elements.photo.value,
+      notes: elements.notes.value,
+      entryId: data.editing.entryId,
+    };
+    const currentEntryId = data.editing.entryId;
+    const indexOfEntriesToReplace = data.entries.findIndex(
+      (v) => v.entryId === currentEntryId,
+    );
+    data.entries[indexOfEntriesToReplace] = newEntry;
+    const $liToAppend = renderEntry(newEntry);
+    if (!$ulForEntries) throw new Error('$ulForEntries not found!');
+    const queryString = '[data-entry-id="' + String(currentEntryId) + '"]';
+    const $liToReplace = $ulForEntries.querySelector(queryString);
+    if (!$liToReplace) throw new Error('$liToReplace not found!');
+    $liToReplace.replaceWith($liToAppend);
+    if (!$entryFormTitle) throw new Error('$entryFormTitle not found!');
+    $entryFormTitle.textContent = 'New Entry';
+    data.editing = null;
+  }
   viewSwap('entries');
   storeData(); // Is after viewSwap because viewSwap changes data.view
   toggleNoEntries();
```

Comment:
I meant that as a statement, not a question. I guess I ought to start with a subject rather than omitting it. I thought the comment ought to be made to warn potential future editors not to mess that up.